### PR TITLE
EG-956, page builder supports Calculation and Choice stanzas, pages now graphs rather than sequences

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/models/ocelot/stanzas/ValueStanza.scala [hunk 1]
```diff
@@ -40,6 +40,7 @@ object Value {
 }
 
 case class ValueStanza(values: List[Value], override val next: Seq[String], stack: Boolean) extends Stanza {
+  override val visual: Boolean = false
```
Comment (LesWilson) [inline app/models/ocelot/stanzas/ValueStanza.scala:43]: As discussed, can we define a NonVisualStanza extending Stanza and setting visual = false.  Then any non-visual stanza extends this new stanza so they don't have to override visual attribute

File: app/models/ocelot/errors/GuidanceErrors.scala [hunk 2]
```diff
@@ -37,6 +37,7 @@ case class PhraseNotFound(id: String, index: Int) extends FlowError
 case class LinkNotFound(id: String, index: Int) extends FlowError
 case class DuplicatePageUrl(id: String, url: String) extends FlowError
 case class MissingWelshText(id: String, index: String, english: String) extends FlowError
+case class VisualStanzasAfterQuestion(id: String) extends FlowError
```
Comment (LesWilson) [inline app/models/ocelot/errors/GuidanceErrors.scala:40]: Could we make this generic to cover all input types - so Question or new InputStanza being developed?  Guess depends if value in identifying it as a question or input stanza

Comment (cmrhmrc) [reply app/models/ocelot/errors/GuidanceErrors.scala:40]: You could include visual stanzas after input fields. Questions are a bit different as GDS makes the question text bethe H1 on the page and the visual stanzas before the question are wrapped in the question, so the question becomes the whole page and no visual stuff can follow.

File: test/services/PageBuilderSpec.scala [hunk 3]
```diff
@@ -878,29 +847,33 @@ class PageBuilderSpec extends BaseSpec with ProcessJson with StanzaHelper {
...
 
-    expectedPageIds.zipWithIndex.foreach {
-      case (id, index) =>
-        indexedPages(index).id shouldBe id
-    }
+    pages.length shouldBe expectedPageIds.length
+
+    pages.forall(p => expectedPageIds.contains(p.id)) shouldBe true
+
+    // expectedPageIds.zipWithIndex.foreach {
```
Comment (LesWilson) [inline test/services/PageBuilderSpec.scala:856]: can this be removed?

**Review by LesWilson — CHANGES_REQUESTED** [review]
A few minor changes

File: app/models/ocelot/stanzas/ValueStanza.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/models/ocelot/stanzas/ValueStanza.scala:43]: will add

File: test/services/PageBuilderSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/services/PageBuilderSpec.scala:856]: yes will do

**Review by LesWilson — APPROVED** [review]

**Pull request merged**